internal/security/secretstore/certs.go

```diff
@@ -0,0 +1,220 @@
+		LoggingClient.Info(e.Error())
+	} else if resp.StatusCode != http.StatusOK {
+		e := fmt.Errorf("failed to retrieve the proxy cert pair on path %s with error code %d", cs.certPath, resp.StatusCode)
+		LoggingClient.Error(e.Error())
+		return nil, e
+	}
+
+	if err = json.NewDecoder(resp.Body).Decode(&cc); err != nil {
+		e := fmt.Errorf("Error decoding json response when retrieving proxy cert pair: %s", err.Error())
+		LoggingClient.Error(e.Error())
+		return nil, err
+	}
+
+	return &cc.Pair, nil
+}
+
+func (cs *Certs) AlreadyinStore() (bool, error) {
+	cp, err := cs.getCertPair()
+	if err != nil {
+		return false, err
+	}
+	if len(cp.Cert) > 0 && len(cp.Key) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (cs *Certs) getCertPair() (*CertPair, error) {
+	t, err := GetAccessToken(cs.tokenPath)
+	if err != nil {
+		return &CertPair{"", ""}, err
+	}
+	cp, err := cs.retrieve(t)
+	if err != nil {
+		return &CertPair{"", ""}, err
+	}
+	return cp, nil
+}
+
+func (cs *Certs) ReadFrom(certPath string, keyPath string) (*CertPair, error) {
+	certPEMBlock, err := ioutil.ReadFile(certPath)
+	if err != nil {
+		return nil, err
+	}
+	cert := string(certPEMBlock)
+
+	keyPEMBlock, err := ioutil.ReadFile(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	key := string(keyPEMBlock)
+
+	cp := CertPair{
+		Cert: cert,
+		Key:  key,
+	}
+
+	return &cp, nil
+}
+
+func GetAccessToken(filename string) (string, error) {
+	a := auth{}
+	raw, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return "", err
+	}
+	err = json.Unmarshal(raw, &a)
+	if err != nil {
+		return "", err
+	}
+
+	return a.Token, nil
+}
+
+func (cs *Certs) UploadToStore(cp *CertPair) error {
+	t, err := GetAccessToken(cs.tokenPath)
+	if err != nil {
+		return err
+	}
+
+	LoggingClient.Info("trying to upload the proxy cert pair into secret store")
+	jsonBytes, err := json.Marshal(cp)
+	body := bytes.NewBuffer(jsonBytes)
+
+	url, err := cs.certPathUrl()
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, body)
+	if err != nil {
+		e := fmt.Errorf("error creating http request: %v", err.Error())
+		LoggingClient.Error(e.Error())
+		return e
+	}
+
+	req.Header.Set(VaultToken, t)
+	resp, err := cs.client.Do(req)
+	if err != nil {
+		e := fmt.Sprintf("failed to upload the proxy cert pair on path %s with error %s", cs.certPath, err.Error())
+		LoggingClient.Error(e)
+		return err
+	}
+	defer resp.Body.Close()
+
+	if !statusSuccess(resp.StatusCode) {
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		e := fmt.Errorf("failed to load the proxy cert pair to the secret store: %s,%s", resp.Status, string(b))
+		LoggingClient.Error(e.Error())
+		return e
+	}
+
+	LoggingClient.Info("successful on uploading the proxy cert pair into secret store")
+	return nil
+}
+
+func statusSuccess(status int) bool {
+	return status >= 200 && status < 300
```

Comment:
I'm -1 on doing it like this, I'd instead like to see all the http statuses handled individually rather than wrapping them all up inside a range of 100 different status codes. For example if for some reason kong reports `208: Already Reported` I don't think we should just swallow that condition silently since Kong clearly probably shouldn't be responding like that

note that an easy way to do this usually is with a switch with all the success conditions enumerated in a single case:

```go
switch code {
case http.StatusOK, http.StatusAccepted:
    // do the right thing
case http.ImATeapot:
    // then don't make coffee
}
```

also please use the http status code constants instead of the numbers